src/main/java/com/fnz/TimeTracking/controller/AuthController.java

```diff
@@ -1,43 +1,60 @@
 package com.fnz.TimeTracking.controller;
 import com.fnz.TimeTracking.model.Utilisateur;
+import com.fnz.TimeTracking.repository.UtilisateurRepository;
 import com.fnz.TimeTracking.service.AuthService;
 import com.fnz.TimeTracking.service.JwtUtil;
+import com.fnz.TimeTracking.service.UtilisateurService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.multipart.MultipartFile;
+
 import java.util.Collections;
 import java.util.Map;
 
 @RestController
 @RequestMapping("/auth")
-public class AuthController
-{
-    @Autowired
-    private AuthService authService;
+public class AuthController {
+
+    private final AuthService authService;
+    private final JwtUtil jwtUtil;
+    private final UtilisateurRepository utilisateurRepository;
```

Comment:
I don't like having both the service and the repository injected in the controller.

I'd really prefer creating findUserByEmail Function in the service that uses the repo function.